price-estimator/src/main.rs

```diff
@@ -98,6 +103,47 @@ struct Options {
     /// an estimate and the solver submitting a solution.
     #[structopt(long, env = "EXTRA_ROUNDING_BUFFER_FACTOR", default_value = "2.0")]
     extra_rounding_buffer_factor: f64,
+
+    // These are copies of the same arguments used in the driver for economic viability.
+    // It would be nice if we could avoid copy pasting these. Maybe structopt/clap has a feature
+    // like serde's flatten that would allow us to do this.
+    #[structopt(
+        long,
+        env = "ECONOMIC_VIABILITY_SUBSIDY_FACTOR",
+        default_value = "10.0"
+    )]
+    economic_viability_subsidy_factor: f64,
+    #[structopt(
+        long,
+        env = "ECONOMIC_VIABILITY_MIN_AVG_FEE_FACTOR",
+        default_value = "1.1"
+    )]
+    economic_viability_min_avg_fee_factor: f64,
+    #[structopt(long, env = "MIN_AVG_FEE_PER_ORDER", default_value = "0")]
+    fallback_min_avg_fee_per_order: u128,
+    #[structopt(long, env = "FALLBACK_MAX_GAS_PRICE", default_value = "100000000000")]
+    fallback_max_gas_price: u128,
+    #[structopt(
+        long,
+        env = "ECONOMIC_VIABILITY_STRATEGY",
+        possible_values = &EconomicViabilityStrategy::variants(),
+        default_value
+    )]
+    economic_viability_strategy: EconomicViabilityStrategy,
+}
+
+arg_enum! {
+    #[derive(Debug)]
+    enum EconomicViabilityStrategy {
+        Dynamic,
+        Static,
+    }
+}
+
+impl Default for EconomicViabilityStrategy {
+    fn default() -> Self {
+        Self::Dynamic
+    }
 }
```

Comment:
Moving this into `services-core` will be tricky because of the `clap` dependency... :thinking: Maybe this is something we can achieve with another crate that focuses on deriving things for enums?

**edit**: Just to be clear, I'm fine with this for now, we can address it as a follow up as you mentioned in the description.